File: Compiler/src/compiler/values/StringValue.java
package compiler.values;

import compiler.Value;
import util.CompilationResult;
import util.DataType;
import util.JenaUtil;

public class StringValue extends Value {
    public StringValue(String value) {
        super(value);
    }

    @Override
    public DataType resultDataType() { return DataType.STRING; }

    @Override
    public CompilationResult compile() {
        return new CompilationResult(JenaUtil.genStingVal(value), "", "");
    }
}

File: Compiler/src/compiler/values/StringValue.kt
package compiler.values

import compiler.Operator
import compiler.Value
import util.CompilationResult
import util.DataType
import util.JenaUtil

class StringValue(value: String) : Value(value) {

    override fun resultDataType(): DataType = DataType.String

    override fun compile(): CompilationResult =
        CompilationResult(JenaUtil.genStingVal(value), listOf(""), "")

    override fun clone(): Operator {
        return StringValue(value)
    }
}